src/main/java/org/commonjava/indy/service/archive/ContentRESTResource.java

```diff
@@ -0,0 +1,32 @@
+package org.commonjava.indy.service.archive;
+
+import io.smallrye.mutiny.Uni;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+
+import org.commonjava.indy.service.archive.services.ContentService;
+import org.jboss.resteasy.annotations.jaxrs.PathParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.File;
+
+@Path("/rest")
+public class ContentRESTResource {
+
+    @Inject
+    ContentService.ArchiveFile archiveFiles;
+
+    @GET
+    @Path("/{path: (.*)}")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public Uni<Response> getFile(@PathParam String path) {
+        File nf = new File(archiveFiles.dir, path);
+        Response.ResponseBuilder response = Response.ok((Object) nf);
```

Comment:
Do we need this class cast here? I remember the Response.ok can accept the pojo directly without classcast